### ISSUE TITLE
[Bug]: CodeView Crash

### ISSUE THREAD

**Issue by yz-gm** [AndroidIDEOfficial/AndroidIDE#843]
### What's the expected behavior?

Crash while writing code 

### What version of AndroidIDE you're using?

v2.3.0 (debug builds)

### Relevant log output

```
AndroidIDE crash report
Manufacturer: Xiaomi
Device: M2012K11AC
ABI: arm64-v8a
SDK version: 31
App version: v2.3.0-beta.internal.1+branch.2b8170b07693b7a9f3c6753ab21edbd4ab106a85.commit.2b8170b.timestamp.20230306225504 (230)

 Stacktrace: 
java.lang.NullPointerException: Attempt to read from field 'int io.github.rosemoe.sora.lang.styling.CodeBlock.startLine' on a null object reference
	at io.github.rosemoe.sora.widget.EditorRenderer.drawBlockLines(EditorRenderer.java:1856)
	at io.github.rosemoe.sora.widget.EditorRenderer.drawView(EditorRenderer.java:663)
	at io.github.rosemoe.sora.widget.EditorRenderer.draw(EditorRenderer.java:199)
	at io.github.rosemoe.sora.widget.CodeEditor.onDraw(CodeEditor.java:4132)
	at android.view.View.draw(Unknown Source:78)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:218)
	at android.view.ViewGroup.recreateChildDisplayList(Unknown Source:21)
	at android.view.ViewGroup.dispatchGetDisplayList(Unknown Source:21)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:88)
	at android.view.ViewGroup.recreateChildDisplayList(Unknown Source:21)
	at android.view.ViewGroup.dispatchGetDisplayList(Unknown Source:21)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:88)
	at android.view.ViewGroup.recreateChildDisplayList(Unknown Source:21)
	at android.view.ViewGroup.dispatchGetDisplayList(Unknown Source:21)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:88)
	at android.view.ViewGroup.recreateChildDisplayList(Unknown Source:21)
	at android.view.ViewGroup.dispatchGetDisplayList(Unknown Source:21)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:88)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.drawChild(CoordinatorLayout.java:1312)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at androidx.drawerlayout.widget.DrawerLayout.drawChild(DrawerLayout.java:1478)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:183)
	at android.view.View.draw(Unknown Source:262)
	at android.view.ViewGroup.drawChild(Unknown Source:0)
	at android.view.ViewGroup.dispatchDraw(Unknown Source:267)
	at android.view.View.draw(Unknown Source:88)
	at com.android.internal.policy.DecorView.draw(Unknown Source:0)
	at android.view.View.updateDisplayListIfDirty(Unknown Source:218)
	at android.view.ThreadedRenderer.updateViewTreeDisplayList(Unknown Source:27)
	at android.view.ThreadedRenderer.updateRootDisplayList(Unknown Source:10)
	at android.view.ThreadedRenderer.draw(Unknown Source:7)
	at android.view.ViewRootImpl.draw(Unknown Source:399)
	at android.view.ViewRootImpl.performDraw(Unknown Source:64)
	at android.view.ViewRootImpl.performTraversals(Unknown Source:2719)
	at android.view.ViewRootImpl.doTraversal(Unknown Source:36)
	at android.view.ViewRootImpl$TraversalRunnable.run(Unknown Source:2)
	at android.view.Choreographer$CallbackRecord.run(Unknown Source:20)
	at android.view.Choreographer.doCallbacks(Unknown Source:84)
	at android.view.Choreographer.doFrame(Unknown Source:328)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Unknown Source:11)
	at android.os.Handler.handleCallback(Unknown Source:2)
	at android.os.Handler.dispatchMessage(Unknown Source:4)
	at android.os.Looper.loopOnce(Unknown Source:176)
	at android.os.Looper.loop(Unknown Source:76)
	at android.app.ActivityThread.main(Unknown Source:138)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:306)

```

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by itsaky** [comment]
Update to the latest version of AndroidIDE. The version of AndroidIDE you're using belongs to the `I10_dev` branch which is used for translations. That branch is always the latest development branch.

When you're downloading APKs from GitHub actions, make sure that they belong to the `dev` or `main` branch. If you don't know what this means, stick to using the [release builds](https://github.com/AndroidIDEOfficial/AndroidIDE/releases).

**Comment by MrIkso** [comment]
@itsaky it's fixed https://github.com/Rosemoe/sora-editor/commit/5d88ae4322485b7fcf73599437506ad808b65840 but not released

**Comment by itsaky** [comment]
Sorry, I thought the commit was included in sora-editor `0.21.0`.

Still, the issue must be closed as it is fixed. All we can do is to wait for @Rosemoe to release the next version of sora-editor.
